### PR TITLE
Feat/dv 112/create endpoint to get trip by

### PR DESCRIPTION
**Pull request by Eugene-pas**

### CODE REVIEW

File: API/Controllers/TripController.cs [hunk 1]
```diff
@@ -69,5 +69,13 @@ public async Task<IActionResult> DeleteRouteAsync([FromQuery] TripIdDTO tripIdDT
 
             return Ok();
         }
+
+        [HttpGet("by-id")]
+        [AuthorizeByRole(IdentityRoleNames.Logist)]
+        public async Task<IActionResult> GetTripById(int tripId)
```
Comment (Grasmus) [inline API/Controllers/TripController.cs]: We should always pass data using a DTO

File: API/Controllers/OfferController.cs [hunk 2]
```diff
@@ -55,9 +55,9 @@ public async Task<IActionResult> GetUserOffersAsync([FromQuery] PaginationFilter
 
         [HttpGet("near-route")]
         [AuthorizeByRole(IdentityRoleNames.Logist)]
-        public async Task<IActionResult> GetNearRouteAsync([FromQuery] PaginationFilterDTO paginationFilter, int routeId)
+        public async Task<IActionResult> GetNearRouteAsync(int routeId)
```
Comment (Grasmus) [inline API/Controllers/OfferController.cs]: We should always pass data using a DTO

File: Core/DTO/PointDTO/PointOfferCreateTripDTO.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+﻿namespace Core.DTO
+{
+    public class PointOfferCreateTripDTO
```
Comment (Grasmus) [inline Core/DTO/PointDTO/PointOfferCreateTripDTO.cs:3]: We already have OfferPointCreateTripDTO. Why does it exist?

File: API/Controllers/TripController.cs [hunk 4]
```diff
@@ -69,5 +69,13 @@ public async Task<IActionResult> DeleteRouteAsync([FromQuery] TripIdDTO tripIdDT
...
+        {
+            return Ok(await _tripService.GetTripByIdAsync(tripId));
+        }
+
```
Comment (Grasmus) [inline API/Controllers/TripController.cs]: Remove empty line

File: Core/Helpers/ApplicationProfiles/PointProfiles.cs [hunk 5]
```diff
@@ -24,6 +24,27 @@ public PointProfiles()
...
+                .ForMember(dest => dest.OfferId, opt => opt.MapFrom(point => point.OfferId))
+                .ForMember(dest => dest.Description, opt => opt.MapFrom(point => point.Offer.Description))
+                .ForMember(dest => dest.RelatedTripId, opt => opt.MapFrom(point => point.Offer.RelatedTripId))
+                .ForMember(dest => dest.GoodsWeight, opt => opt.MapFrom(point => point.Offer.GoodsWeight))
+                .ForMember(dest => dest.GoodCategoryName, opt => opt.MapFrom(point => point.Offer.GoodCategory.Name))
+                .ForMember(dest => dest.CreatorRoleName, opt => opt.MapFrom(point => point.Offer.OfferRole.Name))
+
+
+                ;
+
```
Comment (Grasmus) [inline Core/Helpers/ApplicationProfiles/PointProfiles.cs]: Remove empty lines

File: Core/Helpers/ApplicationProfiles/OfferProfiles.cs [hunk 6]
```diff
@@ -33,13 +33,20 @@ public OfferProfiles()
...
                 .ForMember(dest => dest.Longitude, opt => opt.MapFrom(offer => offer.Point.Location.X))
-                .ForMember(dest => dest.CreationDate, opt => opt.MapFrom(offer => offer.CreationDate.ToString("yyyy/MM/dd HH:mm")))
-                .ForMember(dest => dest.Key, opt => opt.MapFrom(offer => offer.Id));
+                .ForMember(dest => dest.Address, opt => opt.MapFrom(offer => offer.Point.Address))
+                .ForMember(dest => dest.Settlement, opt => opt.MapFrom(offer => offer.Point.Settlement))
+                .ForMember(dest => dest.Region, opt => opt.MapFrom(offer => offer.Point.Region))
+                .ForMember(dest => dest.Country, opt => opt.MapFrom(offer => offer.Point.Country))
+                .ForMember(dest => dest.Postcode, opt => opt.MapFrom(offer => offer.Point.Postcode))
+                .ForMember(dest => dest.Order, opt => opt.MapFrom(offer => offer.Point.Order));
+
```
Comment (Grasmus) [inline Core/Helpers/ApplicationProfiles/OfferProfiles.cs]: Remove empty line

**Review by Grasmus — APPROVED** [review]

**Review by AndriiChepeliuk — APPROVED** [review]

**Pull request merged**